web-common/src/features/dashboards/state-managers/loaders/most-recent-explore-state.ts

```diff
@@ -0,0 +1,159 @@
+import type { MetricsExplorerEntity } from "@rilldata/web-common/features/dashboards/stores/metrics-explorer-entity";
+import type { TimeControlState } from "@rilldata/web-common/features/dashboards/time-controls/time-control-store";
+import { convertExploreStateToURLSearchParams } from "@rilldata/web-common/features/dashboards/url-state/convertExploreStateToURLSearchParams";
+import { convertURLSearchParamsToExploreState } from "@rilldata/web-common/features/dashboards/url-state/convertURLSearchParamsToExploreState";
+import {
+  type DashboardTimeControls,
+  TimeComparisonOption,
+  TimeRangePreset,
+} from "@rilldata/web-common/lib/time/types";
+import { DashboardState_ActivePage } from "@rilldata/web-common/proto/gen/rill/ui/v1/dashboard_pb";
+import type {
+  V1ExploreSpec,
+  V1MetricsViewSpec,
+} from "@rilldata/web-common/runtime-client";
+
+const ExploreViewKeys: (keyof MetricsExplorerEntity)[] = [
+  "allMeasuresVisible",
+  "visibleMeasureKeys",
+  "allDimensionsVisible",
+  "visibleDimensionKeys",
+  "leaderboardMeasureName",
+  "sortDirection",
+  "leaderboardContextColumn",
+];
+// Keys that do not need any special handling and can be directly copied over
+const DirectCopyExploreStateKeys: (keyof MetricsExplorerEntity)[] = [
+  "showTimeComparison",
+  ...ExploreViewKeys,
+];
+// Keys that are not defined in certain views
+const ExploreStateNonDefinedInViewKeys: Record<
+  DashboardState_ActivePage,
+  (keyof MetricsExplorerEntity)[]
+> = {
+  [DashboardState_ActivePage.UNSPECIFIED]: [],
+  [DashboardState_ActivePage.DEFAULT]: [],
+  [DashboardState_ActivePage.DIMENSION_TABLE]: [],
+  [DashboardState_ActivePage.TIME_DIMENSIONAL_DETAIL]: ExploreViewKeys,
+  [DashboardState_ActivePage.PIVOT]: ExploreViewKeys,
+};
```

Comment:
I'm having a hard time following these types. Also, this custom logic for various types feels like it's brittle to `ExploreState` type changes.